### PR TITLE
better error messags and build artifact size information

### LINKED ISSUE #2106: Setup.py build message improvements
* Only show BokehJS build output on errors
* report build artifact sizes

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #2106

### CODE REVIEW

**Comment by bryevdv** [comment]
Success produces output like:
```
bryan@laptop (git:master) ~/work/bokeh $ python setup.py develop --build_js
Building BokehJS... Success!

Build artifact sizes:
  - bokeh.js      : 1980.6 kB
  - bokeh.css     : 554.7 KB
  - bokeh.min.js  : 892.4 KB
  - bokeh.min.css : 525.8 KB

Installing Bokeh for development:
  - writing path '/Users/bryan/work/bokeh' to /Users/bryan/anaconda/lib/python3.4/site-packages/bokeh.pth
  - using NEWLY built BokehJS from bokehjs/build
```
Build failure produces output like:
```
bryan@laptop (git:task/2106_bkjsbuild) ~/work/bokeh $ python setup.py develop --build_js
Building BokehJS...  Failed.

ERROR: 'grunt deploy' returned error message:

Running "coffee:compile" (coffee) task
>> SyntaxError: unmatched )
>> In file: src/coffee/renderer/glyph/oval.coffee
>> On line: 6
>>   class OvalView extends Glyph.View )
>>                                     ^ 
Warning: CoffeeScript failed to compile. Use --force to continue.

Aborted due to warnings.
```
Failure to even exceute the build command produces output like:
```
bryan@laptop (git:task/2106_bkjsbuild) ~/work/bokeh $ python setup.py develop --build_js
Building BokehJS...  Failed.

ERROR: subprocess.Popen(['anode_modules/.bin/grunt', 'deploy']) failed to execute:

    [Errno 2] No such file or directory: 'anode_modules/.bin/grunt'

Have you run `npm install` from the bokehjs subdirectory?
For more information, see the Dev Guide:

    http://bokeh.pydata.org/en/latest/docs/dev_guide.html

```

File: setup.py [hunk 1]
```diff
@@ -210,22 +230,35 @@ def build_js():
...
+        msg = proc.stdout.read()
+        print(BUILD_FAIL_MSG % msg.decode('ascii'))
         sys.exit(1)
 
+    print("Success!")
+    print()
+    print("Build artifact sizes:")
+    try:
+        blddir = join("bokehjs", "build")
+        bkjs_size = os.stat(join(blddir, "js", "bokeh.js")).st_size / 2**10
```
Comment (fpliger) [inline setup.py]: I'd prefer using something like:

```Python
def get_file_size(filename): 
    return os.stat(join(blddir, "js", filename)).st_size / 2**10
bkjs_size = get_file_size("bokeh.js")
bkjs_min_size = get_file_size("bokeh.min.js")
...
```

but it's a very personal thing...

**Comment by fpliger** [comment]
Nice improvement! Left a very minor comment. LGTM

**Comment by bryevdv** [comment]
Spurious single test failure. Merging.

**Pull request merged**